server/routes/explore/helpers.py

```diff
@@ -466,3 +467,20 @@ def _set_blocked(err_json: Dict):
   err_json['blocked'] = True
   if err_json.get('debug'):
     err_json['debug']['blocked'] = True
+
+
+def explore_post_body_cache_key():
+  """
+  Builds flask cache key for /detect and /detect-and-fulfill POST
+  requests. Does not cache requests with a contextHistory.
+  """
+  body_object = request.get_json()
+  if "contextHistory" in body_object and body_object["contextHistory"]:
+    # Don't cache requests with a contextHistory by returning None.
+    # All context histories are be unique due to a unique session ID, so caching
```

Comment:
If that's the only reason, it might be worth constructing the key without the session ID in that case?  Have a TODO?

Reply:
Added a todo